nav2_smac_planner/include/nav2_smac_planner/smoother.hpp

```diff
@@ -243,6 +243,8 @@ class Smoother
       theta = atan2(dy, dx);
       path.poses[i].pose.orientation = orientationAroundZAxis(theta);
     }
+    // Populate last point
+    path.poses.back().pose.orientation = orientationAroundZAxis(theta);
```

Comment:
For the Hybrid-A* planner though, I don't think this would be overridden for the final goal orientation this way. I think that needs to be done in the Hybrid-A* code if you want to add this for the benefit of the 2D code.